src/Beberlei/Bundle/MetricsBundle/DependencyInjection/Configuration.php

```diff
@@ -18,21 +18,36 @@ public function getConfigTreeBuilder()
 
         $rootNode
             ->children()
-                ->scalarNode('default')->defaultValue('default')->end()
+                ->scalarNode('default')
+                    ->defaultNull()
+                ->end()
                 ->arrayNode('collectors')
                     ->isRequired()
                     ->useAttributeAsKey('name')
                     ->prototype('array')
                         ->children()
-                            ->scalarNode('type')->isRequired()->end()
-                            ->scalarNode('hostname')->end()
-                            ->scalarNode('server')->end()
-                            ->scalarNode('host')->end()
-                            ->scalarNode('port')->end()
-                            ->scalarNode('file')->end()
-                            ->scalarNode('username')->end()
-                            ->scalarNode('password')->end()
-                            ->scalarNode('connection')->end()
+                            ->scalarNode('type')->end()
+                            ->scalarNode('connection')->defaultNull()->end()
+                            ->scalarNode('file')->defaultNull()->end()
+                            ->scalarNode('host')->defaultNull()->end()
+                            ->scalarNode('password')->defaultNull()->end()
+                            ->scalarNode('port')->defaultNull()->end()
+                            ->scalarNode('prefix')->defaultNull()->end()
+                            ->scalarNode('protocol')->defaultNull()->end()
+                            ->scalarNode('source')->defaultNull()->end()
+                            ->scalarNode('username')->defaultNull()->end()
+                        ->end()
+                        ->validate()
+                            ->ifTrue(function ($v) { return 'librato' === $v['type'] && empty($v['source']); })
```

Comment:
``$v['type']`` could be undefined at this point given ``type`` is not required and does not have a default value